experimental/dds/tree2/src/feature-libraries/contextuallyTyped.ts

```diff
@@ -457,18 +462,41 @@ export function applyTypesFromContext(
 			primary !== undefined,
 			0x4d6 /* array data reported comparable with the schema without a primary field */,
 		);
-		const children = applyFieldTypesFromContext(schemaData, primary.schema, data);
+		const children = applyFieldTypesFromContext(context.schemaData, primary.schema, data);
 		return {
 			value: undefined,
 			type,
 			fields: new Map(children.length > 0 ? [[primary.key, children]] : []),
 		};
 	} else {
+		if (
+			context.globalFieldKeySymbol !== undefined &&
+			data[context.globalFieldKeySymbol] === undefined
+		) {
+			const globalFieldSchema = getFieldSchema(
+				context.globalFieldKeySymbol,
+				context.schemaData,
+				schema,
+			) as GlobalFieldSchema;
+			const globalSchema = globalFieldSchema.schema;
+			const globalTreeSchema = globalSchema.allowedTypes[0] as TreeSchema;
```

Comment:
All tree schema are "global", it's just fields which have local and global schema.

Generally I think these variables have names which aren't very descriptive. I think this is the schema of the child of field you are generating an instance of: "globalTreeSchema" doesn't clarify that very well.

Also assuming there is exactly one allowed type isn't great, and neither is assuming that if it requires a string a UUID is appropriate (unless the API we expose for this documents if fills the provided field with UUID string if in schema an otherwise does nothing.

Reply:
I should note that part of the reason I defined getFieldGenerator in the context so it could be provided by the caller is to avoid having to make code here be specific to particular content or schema. It also results in thos code havine simpler flow control (less branches, switches etc) making it easier to unit tests (the tree generation logic can be tested separate from testing that the generator gets used)

Reply:
updated to use `getFieldGenerator` so the caller can pass in the schema